request/id.go

```diff
@@ -64,7 +65,7 @@ func Parse(id string) (ID, error) {
 	}
 
 	return ID{
-		timestamp: uint64(timestamp),
+		timestamp: uint64(math.Abs(float64(timestamp))),
```

Comment:
I don't quite agree with this solution, as it composes casting over casting. It would resolve the linter error, but not really fix the problem (risk of bad behavior on negative integers)

I believe that a better solution would be

1. Use strconv.ParseInt to pare an int64 instead of int32 (still not unsigned, but int64 to uint64 is a "smaller" difference than int32 to uint64)
2. If the cast fails, or the resulting number is negative, return error.

What do you think?

Reply:
I agree that the proposed solution uses unnecessary casting to float.

I believe a simple `if` would suffice. My suggestion would be:

```
// ...
		timestamp: toUint64(timestamp),
// ...


func toUint64(i int) uint64 {
	if i < 0 {
		panic("cannot cast negative int to uint64")
	}

	return uint64(i)
}
```

Reply:
Forget about what I said. I forgot about ParseUint. 

The correct solution would be:
```
	timestamp, err := strconv.ParseUint(slices[0], 10, 64)
	if err != nil {
		return ID{}, err
	}

	return ID{
		timestamp: timestamp,
		randomID:  slices[1],
	}, nil
```